src/librustc_borrowck/diagnostics.rs

```diff
@@ -293,6 +293,66 @@ let c = &i; // still ok!
 ```
 "##,
 
+E0507: r##"
+You tried to move a borrowed content. Erroneous code example:
+
+```
+use std::cell::RefCell;
+
+struct TheDarkKnight;
+
+impl TheDarkKnight {
+    fn nothing_is_true(self) {}
+}
+
+fn main() {
+    let x = RefCell::new(TheDarkKnight);
+
+    x.borrow().nothing_is_true(); // error: cannot move out of borrowed content
+}
+```
+
+Here, the `nothing_is_true` method takes the ownership of `self`. However, `self`
+cannot be moved because this is `RefCell` which does have the ownership. To fix
```

Comment:
because `.borrow()` only provides an `&TheDarkKnight`, which is a borrow of the content owned by the `RefCell`.